### ISSUE TITLE
java.lang.AbstractMethodError: javax.ws.rs.core.UriBuilder.uri(Ljava/lang/String;)Ljavax/ws/rs/core/UriBuilder;

### ISSUE THREAD

**Issue by bhagyashri10** [confluentinc/kafka-streams-examples#93]
I am using eclipse IDE to run **KafkaMusicExample:**
https://github.com/confluentinc/kafka-streams-examples/blob/4.0.0-post/src/main/java/io/confluent/examples/streams/interactivequeries/kafkamusic/KafkaMusicExample.java

I got the following error at 5th step:

**5) Use your browser to hit the REST endpoint of the app instance you started in step 3 to query the state managed by this application.**

example:
List all running instances of this application
http://localhost:7070/kafka-music/instances
  

```
[2018-03-21 15:54:39,472] WARN Error for /kafka-music/instances (org.eclipse.jetty.servlet.ServletHandler:667)
java.lang.AbstractMethodError: javax.ws.rs.core.UriBuilder.uri(Ljava/lang/String;)Ljavax/ws/rs/core/UriBuilder;
	at javax.ws.rs.core.UriBuilder.fromUri(UriBuilder.java:119)
	at org.glassfish.jersey.servlet.ServletContainer.service(ServletContainer.java:291)
	at org.glassfish.jersey.servlet.ServletContainer.service(ServletContainer.java:221)
	at org.eclipse.jetty.servlet.ServletHolder.handle(ServletHolder.java:808)
	at org.eclipse.jetty.servlet.ServletHandler.doHandle(ServletHandler.java:587)
	at org.eclipse.jetty.server.session.SessionHandler.doHandle(SessionHandler.java:221)
	at org.eclipse.jetty.server.handler.ContextHandler.doHandle(ContextHandler.java:1127)
	at org.eclipse.jetty.servlet.ServletHandler.doScope(ServletHandler.java:515)
	at org.eclipse.jetty.server.session.SessionHandler.doScope(SessionHandler.java:185)
	at org.eclipse.jetty.server.handler.ContextHandler.doScope(ContextHandler.java:1061)
	at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:141)
	at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:97)
	at org.eclipse.jetty.server.Server.handle(Server.java:499)
	at org.eclipse.jetty.server.HttpChannel.handle(HttpChannel.java:310)
	at org.eclipse.jetty.server.HttpConnection.onFillable(HttpConnection.java:257)
	at org.eclipse.jetty.io.AbstractConnection$2.run(AbstractConnection.java:540)
	at org.eclipse.jetty.util.thread.QueuedThreadPool.runJob(QueuedThreadPool.java:635)
	at org.eclipse.jetty.util.thread.QueuedThreadPool$3.run(QueuedThreadPool.java:555)
	at java.lang.Thread.run(Unknown Source)
```

**Comment by mjsax** [comment]
Sound like a dependency issue, ie, some dependent lib might be pulled in with wrong version?

**Comment by bhagyashri10** [comment]
I have use the following JARS:

```
jersey-bean-validation-2.19.jar
jersey-bundle-1.9.jar
jersey-client-2.19.jar
jersey-common-2.19.jar
jersey-container-jetty-http-2.19.jar
jersey-container-servlet-2.19.jar
jersey-container-servlet-core-2.19.jar
jersey-entity-filtering-2.19.jar
jersey-guava-2.19.jar
jersey-media-jaxb-2.19.jar
jersey-media-json-jackson-2.19.jar
jersey-media-multipart-2.13.jar
jersey-server-2.19.jar
jersey-test-framework-core-2.19.jar
jersey-test-framework-provider-jetty-2.19.jar

jetty-continuation-9.2.12.v20150709.jar
jetty-http-9.2.12.v20150709.jar
jetty-io-9.2.12.v20150709.jar
jetty-security-9.2.12.v20150709.jar
jetty-server-9.2.12.v20150709.jar
jetty-servlet-9.2.12.v20150709.jar
jetty-servlets-9.2.12.v20150709.jar
jetty-util-9.2.12.v20150709.jar

javax.annotation-api-1.2.jar
javax.el-2.2.4.jar
javax.el-api-2.2.4.jar
javax.inject-2.4.0-b25.jar
javax.servlet-api-3.1.0.jar
javax.ws.rs-api-2.0.1.jar
```

**Comment by bhagyashri10** [comment]
Problem solved by removing jersey-bundle-1.9.jar and add following code in web.xml file.

```
<servlet>
	<servlet-name>Jersey REST Service</servlet-name>
	<servlet-class>
	  	org.glassfish.jersey.servlet.ServletContainer
	</servlet-class>
	<init-param>
	    <param-name>jersey.config.server.provider.packages</param-name>
	    <param-value>sample.hello.resources</param-value>
	</init-param>
	<load-on-startup>1</load-on-startup>
  </servlet>
  <servlet-mapping>
	  <servlet-name>Jersey REST Service</servlet-name>
	  <url-pattern>/*</url-pattern>
  </servlet-mapping>
```

**Comment by bhagyashri10** [comment]
removed jersey-bundle-1.9.jar
and added following code in web.xml solved the issue.

```
<servlet>
	<servlet-name>Jersey REST Service</servlet-name>
	<servlet-class>
	  	org.glassfish.jersey.servlet.ServletContainer
	</servlet-class>
	<init-param>
	    <param-name>jersey.config.server.provider.packages</param-name>
	    <param-value>sample.hello.resources</param-value>
	</init-param>
	<load-on-startup>1</load-on-startup>
  </servlet>
  <servlet-mapping>
	  <servlet-name>Jersey REST Service</servlet-name>
	  <url-pattern>/*</url-pattern>
  </servlet-mapping>
```
